File: opex-frontend/src/types/transfer.ts
// src/types/transfer.ts (수정 후 최종 형태)

export interface TransferRequest {
    from_proj_id: string;
    to_proj_id: string;
    transfer_amount: number;
    transfer_yyyymm: string;
    reason?: string;
    transferred_by?: string; 
}

export interface TransferLog {
    transfer_id: number;
    from_proj_id: string;
    to_proj_id: string;
    transfer_amount: number;
    transfer_yyyymm: string;
    transferred_at: string; // string으로 유지
    reason?: string;
}

File: opex-backend/app/schemas/transfer.py
# app/schemas/transfer.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class TransferRequest(BaseModel):
    from_proj_id: str
    to_proj_id: str
    transfer_amount: int
    transfer_yyyymm: str
    reason: Optional[str] = None
    transferred_by: Optional[str] = "admin" # 임시 사용자 ID

class TransferLog(BaseModel):
    transfer_id: int
    from_proj_id: str
    to_proj_id: str
    transfer_amount: int
    transfer_yyyymm: str
    transferred_at: datetime

    class Config:
        from_attributes = True